### PR TITLE
Synu.copy fails when copying a file with READ permissions

### PR DESCRIPTION
**Pull request by thomasyu888**
```
  File "/home/ubuntu/.local/lib/python3.6/site-packages/synapseutils/copy.py", line 204, in _copyRecursive
    skipCopyAnnotations=skipCopyAnnotations, **kwargs)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/synapseutils/copy.py", line 207, in _copyRecursive
    **kwargs)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/synapseutils/copy.py", line 253, in _copyFolder
    _copyRecursive(syn, ent['id'], newFolder.id, mapping, skipCopyAnnotations=skipCopyAnnotations, **kwargs)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/synapseutils/copy.py", line 210, in _copyRecursive
    setProvenance=setProvenance, skipCopyAnnotations=skipCopyAnnotations)
  File "/home/ubuntu/.local/lib/python3.6/site-packages/synapseutils/copy.py", line 307, in _copyFile
    createdBy = fileHandle['createdBy']
TypeError: 'NoneType' object is not subscriptable

```

### CODE REVIEW

**Comment by thomasyu888** [comment]
https://sagebionetworks.jira.com/browse/SYNPY-881

**Comment by kimyen** [comment]
@thomasyu888 Thank you for submitting the fix!
Please also add the following:
* A unit test that fail without the fix, and produce the same error. This test shall pass with your changes.
* Updating the documentation on what are the requirements to copy a file (what permissions are requires, ...)

File: synapseutils/copy.py [hunk 1]
```diff
@@ -304,32 +307,35 @@ def _copyFile(syn, entity, destinationId, version=None, updateExisting=False, se
...
-        newdataFileHandleId = copyResult['newFileHandle']['id']
-
-    new_ent = File(dataFileHandleId=newdataFileHandleId,  name=ent.name, parentId=destinationId)
-    # Set annotations here
-    if not skipCopyAnnotations:
-        new_ent.annotations = ent.annotations
-    # Store provenance if act is not None
-    if act is not None:
-        new_ent = syn.store(new_ent, activity=act)
+    #if the filehandle is none, it probably means you only have READ access to the entity
```
Comment (thomasyu888) [inline synapseutils/copy.py]: Revert this change

**Comment by thomasyu888** [comment]
New PR https://github.com/Sage-Bionetworks/synapsePythonClient/pull/637
